Library/Homebrew/extend/fileutils.rb

```diff
@@ -6,42 +6,83 @@
 # @see http://ruby-doc.org/stdlib-1.8.7/libdoc/fileutils/rdoc/FileUtils.html Ruby's FileUtils API
 module FileUtils
   # Create a temporary directory then yield. When the block returns,
-  # recursively delete the temporary directory.
-  def mktemp(prefix = name)
-    prev = pwd
-    tmp  = Dir.mktmpdir(prefix, HOMEBREW_TEMP)
-
-    # Make sure files inside the temporary directory have the same group as the
-    # brew instance.
-    #
-    # Reference from `man 2 open`
-    # > When a new file is created, it is given the group of the directory which
-    # contains it.
-    group_id = if HOMEBREW_BREW_FILE.grpowned?
-      HOMEBREW_BREW_FILE.stat.gid
-    else
-      Process.gid
+  # recursively delete the temporary directory. Passing opts[:retain]
+  # or calling `do |staging| ... staging.retain!` in the block will suppress
+  # the deletion for the topmost mktemp on the call stack, if any.
```

Comment:
The piece about the topmost `mktemp` and the call stack no longer applies, right?